project/accounts/views.py

```diff
@@ -24,20 +23,41 @@ def form_valid(self, form):
         return redirect(self.success_url)
 
 
-@login_required
-def user_profile(request):
-    if request.method == "POST":
-        user_form = UpdateUserForm(request.POST, instance=request.user)
+# @login_required
+# def user_profile(request):
+#     if request.method == "POST":
+#         user_form = UpdateUserForm(request.POST, instance=request.user)
+
+#         if user_form.is_valid():
+#             user_form.save()
+
+#             messages.success(request, _("Profile updated successfully"))
+
+#             return redirect(to="user-profile")
+#         else:
+#             messages.error(request, _("Profile form isn't valid"))
+#     else:
+#         user_form = UpdateUserForm(instance=request.user)
 
-        if user_form.is_valid():
-            user_form.save()
+#     return render(request, "accounts/profile.html", {"user_form": user_form})
 
-            messages.success(request, _("Profile updated successfully"))
 
-            return redirect(to="user-profile")
-        else:
-            messages.error(request, _("Profile form isn't valid"))
-    else:
-        user_form = UpdateUserForm(instance=request.user)
+class UserProfileUpdateView(FormView):
+    form_class = UpdateUserForm
+    template_name = "accounts/profile.html"
+    success_url = reverse_lazy("user-profile")
 
-    return render(request, "accounts/profile.html", {"user_form": user_form})
+    def get_initial(self):
+        initial = super().get_initial()
+        initial["display_name"] = self.request.user.display_name
+        initial["email"] = self.request.user.email
+        return initial
+
+    def get_context_data(self, **kwargs):
+        context = super(UserProfileUpdateView, self).get_context_data(**kwargs)
+        context["user_form"] = self.get_form()
+        return context
```

Comment:
Let's go ahead and rename the form in the template to `form` instead of `user_form`, so this call to `get_context_data` won't be necessary.